etl/helpers.py

```diff
@@ -591,6 +627,15 @@ def load_config(self, filename: Optional[str] = None, path: Optional[str | Path]
         elif path is None:
             path = self.config_path
         config = catalog.utils.dynamic_yaml_to_dict(catalog.utils.dynamic_yaml_load(path))
+
+        # Extract all indicators mentioned in the config.
+        indicators = extract_indicators_from_config(config=config)
+        # List their corresponding datasets.
+        datasets = sorted(set(["data://" + "/".join(indicator.split("/")[:-1]) for indicator in indicators]))
+        # Ensure that their datasets are included as dependencies of the current step.
+        if not set(datasets) <= self.dependencies:
+            raise NoMatchingStepsAmongDependencies(step_name=self.step_name)
```

Comment:
I think this won't work in cases where the catalog path is not complete (i.e. use `table_name#column`). Note that that is often the case (and actually the recommended way).

There is a sanity check on this later in `create_explorer`, which tries to expand the catalog paths so they are complete based on the given dependencies (see function `process_views`). It'll fail if it can't do it (e.g. no step with that table name).

Hence, I suggest to drop this bit